### PR TITLE
Use pie slice color for UserCountGraph chart line when pie slice is selected

### PR DESCRIPTION
**Pull request by johnPhillips**
## Summary

Addresses issue #2625

## Checklist

- [x] My code is tested and passes existing unit tests.
- [ ] My code has an appropriate set of unit tests which all pass.
- [ ] My code is backward-compatible with WordPress 4.7 and PHP 5.6.
- [ ] My code follows the [WordPress](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) coding standards.
- [ ] My code has proper inline documentation.
- [ ] I have added a QA Brief on the issue linked above.
- [x] I have signed the Contributor License Agreement (see <https://cla.developers.google.com/>).

### CODE REVIEW

File: assets/js/modules/analytics/components/dashboard/DashboardAllTrafficWidgetV2/UserCountGraph.js [hunk 1]
```diff
@@ -77,6 +78,9 @@ export default function UserCountGraph( { dimensionName, dimensionValue } ) {
 	const { startDate, endDate } = useSelect( ( select ) => select( CORE_USER ).getDateRangeDates( {
 		offsetDays: DATE_RANGE_OFFSET,
 	} ) );
+	const { getValue } = useSelect( ( select ) => select( CORE_FORMS ) );
+	const dimensionColor = getValue( FORM_ALL_TRAFFIC_WIDGET, 'dimensionColor' );
+	const graphLineColor = dimensionColor || '#1a73e8';
```
Comment (eugene-manuilov) [inline assets/js/modules/analytics/components/dashboard/DashboardAllTrafficWidgetV2/UserCountGraph.js:83]: Let's write it in a single line:

```suggestion
	const graphLineColor = useSelect( ( select ) => select( CORE_FORMS ).getValue( FORM_ALL_TRAFFIC_WIDGET, 'dimensionColor' ) || '#1a73e8' );
```

File: assets/js/modules/analytics/components/dashboard/DashboardAllTrafficWidgetV2/DimensionTabs.js [hunk 2]
```diff
@@ -58,7 +58,7 @@ export default function DimensionTabs( { dimensionName } ) {
 	const activeTab = tabs.findIndex( ( v ) => v.dimensionName === dimensionName );
 
 	const handleTabUpdate = useCallback( ( index ) => {
-		setValues( FORM_ALL_TRAFFIC_WIDGET, { dimensionName: tabs[ index ].dimensionName, dimensionValue: '' } );
+		setValues( FORM_ALL_TRAFFIC_WIDGET, { dimensionName: tabs[ index ].dimensionName, dimensionValue: '', dimensionColor: '' } );
```
Comment (eugene-manuilov) [inline assets/js/modules/analytics/components/dashboard/DashboardAllTrafficWidgetV2/DimensionTabs.js:61]: Let's split it into multiple lines to make it easier to read:

```suggestion
		setValues( FORM_ALL_TRAFFIC_WIDGET, {
			dimensionName: tabs[ index ].dimensionName,
			dimensionValue: '',
			dimensionColor: '',
		} );
```

File: assets/js/modules/analytics/components/dashboard/DashboardAllTrafficWidgetV2/UserCountGraph.js [hunk 3]
```diff
@@ -186,7 +190,7 @@ export default function UserCountGraph( { dimensionName, dimensionValue } ) {
 					},
 					series: {
 						0: { targetAxisIndex: 0, lineWidth: 0 },
-						1: { color: '#1a73e8', lineWidth: 3, targetAxisIndex: 1 },
+						1: { color: graphLineColor, lineWidth: 3, targetAxisIndex: 1 },
```
Comment (eugene-manuilov) [inline assets/js/modules/analytics/components/dashboard/DashboardAllTrafficWidgetV2/UserCountGraph.js:193]: Same here, let's split it into multiple lines for readability purposes:

```suggestion
						1: {
							color: graphLineColor,
							lineWidth: 3,
							targetAxisIndex: 1,
						},
```

**Review by eugene-manuilov — CHANGES_REQUESTED** [review]
Awesome! Thanks, @johnPhillips! All looks good to me, just added a few nit-picks to adjust formatting a little bit. Could you please fix it and I'll approve this PR?

**Comment by johnPhillips** [comment]
Thanks for the review @eugene-manuilov!
I made the changes you suggested, let me know if there's anything else you can see 👍

**Review by eugene-manuilov — APPROVED** [review]
Awesome!

**Review by felixarntz — APPROVED** [review]
Awesome!

**Pull request merged**